include/odp/api/spec/ipsec.h

```diff
@@ -1210,16 +1256,22 @@ typedef struct odp_ipsec_status_t {
  *
  * The operation does packet transformation according to IPSEC standards (see
  * e.g. RFC 4302 and 4303). Resulting packets are well formed, reconstructed
- * original IP packets, with IPSEC headers removed and valid header field values
- * restored. The amount and content of packet data before the IP header is
- * undefined.
+ * original IP (or TFC dummy) packets, with IPSEC headers removed and valid
+ * header field values restored. The amount and content of packet data before
+ * the IP header is undefined. TFC padding may follow the IP packet payload,
+ * in which case packet length is larger than protocol headers indicate.
+ * TFC dummy packets have both IPv4 and IPv6 flags cleared, although L3 offset
+ * is set also for those.
  *
  * Each successfully transformed packet has a valid value for these metadata
  * regardless of the inner packet parse configuration
  * (odp_ipsec_inbound_config_t):
- * - L3 offset: Offset to the first byte of the (outmost) IP header
- * - pktio:     For inline IPSEC processed packets, original packet input
- *              interface
+ * - l3_offset:  Offset to the first byte of the original IP (or TFC dummy)
+ *               packet
+ * - has_ipv4/6: Specifies if the original packet is IPv4 or IPv6. For tunnel
+ *               mode TFC dummy packets neither flag is set.
```

Comment:
I assume  this is referring to the `odp_packet_has_ipv4()` and `odp_packet_has_ipv6()` accessor functions? Since these bits are only accessible via these functions, this forces applications to play a guessing game with them and their L4 counterparts. Might it be better to consider having `odp_packet_l3_proto()` and `odp_packet_l4_proto()` functions?

Reply:
Application needs to anyway check if packet is v4 or v6. Today it's checking that from first byte of the packet. With TFC tunnel mode, first byte is garbage that cannot be used any more. So, application uses these APIs instead in if - else if - else fashion. There's no more guessing than before.

Reply:
OK